### PR TITLE
chore: upgrade to cosmos-sdk v1.19.0-sdk-v0.46.16

### PR DESCRIPTION
**Pull request by rootulp**
Upgrades to https://github.com/celestiaorg/cosmos-sdk/releases/tag/v1.19.0-sdk-v0.46.16 to unblock https://github.com/celestiaorg/celestia-app/pull/2860

### CODE REVIEW

**Comment by rootulp** [comment]
Identified the issue by reproing locally

```
$ go build -tags ledger -o celestia-appd ../../cmd/celestia-appd
# github.com/cosmos/cosmos-sdk/crypto/ledger
/Users/rootulp/go/pkg/mod/github.com/celestiaorg/cosmos-sdk@v1.19.0-sdk-v0.46.16/crypto/ledger/ledger_real.go:18:10: cannot use device (variable of type *ledger_cosmos_go.LedgerCosmos) as SECP256K1 value in return statement: *ledger_cosmos_go.LedgerCosmos does not implement SECP256K1 (wrong type for method SignSECP256K1)
		have SignSECP256K1([]uint32, []byte, byte) ([]byte, error)
		want SignSECP256K1([]uint32, []byte) ([]byte, error)
```

Even though ledger-cosmos-go was downgraded to v0.12.4 [here](https://github.com/celestiaorg/cosmos-sdk/pull/357/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R21) it wasn't downgraded in this PR, yet...

File: go.mod [hunk 1]
```diff
@@ -44,7 +44,7 @@ require (
 	github.com/bits-and-blooms/bitset v1.7.0 // indirect
 	github.com/consensys/bavard v0.1.13 // indirect
 	github.com/consensys/gnark-crypto v0.12.1 // indirect
-	github.com/cosmos/ledger-cosmos-go v0.13.2 // indirect
+	github.com/cosmos/ledger-cosmos-go v0.12.4 // indirect
```
Comment (rootulp) [inline go.mod:47]: [for reviewers] this was downgraded to match the version in celestiaorg/cosmos-sdk which matches the latest v0.46.x release of cosmos/cosmos-sdk.

Ref: 
1. https://github.com/celestiaorg/cosmos-sdk/pull/357/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R21
2. https://github.com/cosmos/cosmos-sdk/blob/7799bba7bc889288607576aa11655b5fc31a2da9/go.mod#L21

**Comment by rootulp** [comment]
I'm happy we have a Ledger test that caught this! We still need more Ledger tests (i.e. for the pre-built binaries).

**Review by cmwaters — APPROVED** [review]
Don't mind merging this but It will probably be immediately overwritten by v1.20.0

**Comment by rootulp** [comment]
This PR is no longer necessary b/c https://github.com/celestiaorg/celestia-app/pull/2860#discussion_r1402362245

**Pull request closed without merging**